pkg/sqlite/load.go

```diff
@@ -1435,35 +1476,94 @@ func (s *sqlLoader) DeprecateBundle(path string) error {
 		return err
 	}
 
+	// Traverse replaces chain, keeping track of last encountered branch point.
+	// 1.1.2 (1.1.x) <----- 1.1.1 (1.1.x, alpha) <--- 1.1.0 (1.1.x, alpha)
+	// 1.1.3-rc (alpha) <---/
+	// At a branch point, a bundle will have a different channel list to the one it gets replaced by.
+	// Removing the entire tail for deprecating 1.1.3-rc will make the replaces value for 1.1.2 invalid,
+	// so all bundles till the last branch point for the tail bundle set must only be deprecated, not removed.
+	//
+	// multiple bundles replacing a deprecated bundle with a skipped edge requires each branch to be handled separately,
+	// 1.1.2 (alpha,1.1.x) <------------- 1.1.0 (alpha,stable)
+	//   ^   ^-...- 1.1.2-rc1 (alpha) <---/    /
+	//    \-..- 1.1.1 (1.1.x,stable,alpha) <--/
+	// deprecating 1.1.2 will deprecate 1.1.1 and 1.1.0 but truncate 1.1.2-rc-1 in the example above.
+
```

Comment:
Question: When we only set deprecate what is the OLM behaviour? Is that only ignored in the upgrade?
 What is the OLM behaviour if the head of the channel is flagged as deprecated? Would I still be able to install and see the bundle and channel?